### DISCUSSION TITLE
How to blend between image pyramids at different zooms (like in powers of ten)

### DISCUSSION THREAD

**Discussion by bennlich** [openseadragon/openseadragon#2648] [Q&A]
Hello! I am wondering how I might go about blending between different image pyramids at different zooms, kind of like in powers of ten (https://www.youtube.com/watch?v=0fKBhvDjuy0). Or maybe there is a way to construct a single blended image pyramid?

I guess at a high level I'm looking for a way to control which details of my image appear at different zoom levels. Does that make sense? In the interactive mapping world, this is sometimes achieved by specifying a range of zooms (or a range of opacities across zooms) for which a particular data layer is visible.

**Comment by iangilman** [comment]
Cool! On the original Seadragon team at Microsoft, we called this "semantic zoom" (where the content changes at different zoom levels). I don't know that a lot of people have done it with OSD, but it's certainly possible. 

One approach would be to construct your DZI in such a way that each level of the pyramid has different content. There aren't necessarily tools to do this directly, but it's a pretty simple format, so it shouldn't be hard. I guess the easiest would be to make different images (all the same pixel dimensions) and convert them all as separate DZIs and then swap the level folders around. 

This is akin to what happens in mapping, where each level of tiles has different sized text burned into them, for instance. The downside to that you don't have a lot of control over when you transition from one level to another. 

The other approach would be to create them as separate image pyramids and stack them on top of each other in the viewer. Pay attention to the viewer's zoom level and change the opacity of the different images as appropriate. This should be pretty straightforward using the [multi-image API](https://openseadragon.github.io/examples/multi-image/).
